src/components/category-card/category-card.tsx

```diff
@@ -0,0 +1,44 @@
+import { useTranslations } from 'next-intl';
+import Link from 'next/link';
+import React from 'react';
+//PLEASE CHECK USEtRANSLATION AND LET US KNOW
+interface CategoryCardProps {
+	title: string;
+	icon?: React.JSX.ElementType;
+	href?: string;
+	asDiv?: boolean;
+	numberOfProducts: number;
+	onClickAction?: () => void;
+}
+
+const componentTypeDecider = (asDiv: boolean, props: Record<string, any>) => {
+	if (asDiv) return 'div';
+	if (props.href) return Link;
+	return 'div';
+};
+
+const CategoryCard: React.FC<CategoryCardProps> = ({
+	asDiv = false,
+	...props
+}) => {
+	const Component = componentTypeDecider(asDiv, props);
+	// const t = useTranslations();
+	return (
+		<div className="border-2 border-gray-100 w-42 h-52 md:w-52 md:h-56 rounded-2xl m-4 ">
+			<Component
+				href={props.href ?? ''}
+				onClick={props.onClickAction}
+				className="flex items-center justify-center flex-col text-center gap-4 mt-8"
+			>
+				{props.icon && <props.icon className="w-20 h-20" />}
+				<h2 className="vm-text-eh6">{props.title}</h2>
+				<p className="vm-text-p1-regular">
+					{props.numberOfProducts}
+					{/* {`${'common.products'}`} */}
```

Comment:
The problem this wasn't working because you didn't call the function.
`t` is a function where you pass a key as a parameter and it will return you a string depending on the current locale user selected.

```suggestion
					{t('common.products')}
```